pages/entryListPage.js

```diff
@@ -11,40 +14,87 @@ export default function EntryListHome() {
     }
   );
 
+  const [editingEntry, setEditingEntry] = useState(null);
+
   const addJournalEntry = (newJournalEntry) => {
     setJournalEntries((oldJournalEntries) => [
       { id: nanoid(), ...newJournalEntry },
       ...oldJournalEntries,
     ]);
   };
+
+  function handleEditJournalEntry(editedEntry) {
+    setJournalEntries((oldJournalEntries) =>
+      oldJournalEntries.map((journalEntry) => {
+        if (journalEntry.id === editingEntry.id) {
+          return {
+            ...journalEntry,
+            ...editedEntry,
```

Comment:
Since your `editedEntry` contains all fields you need for your Entry you don't need to spread it here over the old journalEntry. The problem is, that you generate your id with nanoid inside your Form. So every time you edit your form a new id is generated. You need to remove the id generation from the form. Since you create an ID in the addJournalEntry anyways you can just remove it from the FormComponent. Then you can write here:
```
return {
  id: journalEntry.id,
  ...editedEntry
}
```